File: app/src/main/java/com/davismiyashiro/expenses/view/addparticipant/ParticipantInterfaces.java
package com.davismiyashiro.expenses.view.addparticipant;

import com.davismiyashiro.expenses.datatypes.Participant;

/**
 * Created by Davis on 25/08/2016.
 */
public interface ParticipantInterfaces {
    interface View {
        void setParticipantName (String name);
        void setParticipantEmail (String email);
        void setParticipantPhone (String phone);
        void showNameError (boolean value);
        void showEmailError(boolean value);
        void showPhoneError(boolean value);
        void onSuccess ();
    }

    interface UserActionsListener {
        void validateInputFields (String name, String phone, String email);
        void addParticipant(Participant participant);
        void updateParticipant (Participant participant);
    }
}

File: app/src/main/java/com/davismiyashiro/expenses/view/addparticipant/ParticipantInterfaces.kt
package com.davismiyashiro.expenses.view.addparticipant

import com.davismiyashiro.expenses.datatypes.Participant

/**
 * Created by Davis on 25/08/2016.
 */
interface ParticipantInterfaces {
    interface View {
        fun setParticipantName(name: String)
        fun setParticipantEmail(email: String)
        fun setParticipantPhone(phone: String)
        fun showNameError(value: Boolean)
        fun showEmailError(value: Boolean)
        fun showPhoneError(value: Boolean)
        fun onSuccess()
    }

    interface UserActionsListener {
        fun validateInputFields(name: String, phone: String, email: String)
        fun addParticipant(participant: Participant)
        fun updateParticipant(participant: Participant)
    }
}
